### PR TITLE
TMDM-13102 Validate filename in MDMServerMessageConsole

### PR DESCRIPTION
**Pull request by sbliu**
**What is the current behavior?** (You should also link to an open issue here)
https://jira.talendforge.org/browse/TMDM-13102
There is a security risk as the fileName could contain ".." characters to break out of the output file path.

**What is the new behavior?**
For security risk reason, if extracted filename contains invalid char "..", set default file name to value that according to corresponding to current log.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.mdm.workbench.serverexplorer/src/main/java/org/talend/mdm/workbench/serverexplorer/console/MDMServerMessageConsole.java [hunk 1]
```diff
@@ -513,6 +513,9 @@ private void download(String dirPath, boolean needOpen, IProgressMonitor monitor
             if (HTTP_STATUS_OK == code) {
 
                 String fileName = getFileName(response);
+                if (fileName.contains("..")) {
```
Comment (coheigea) [inline main/plugins/org.talend.mdm.workbench.serverexplorer/src/main/java/org/talend/mdm/workbench/serverexplorer/console/MDMServerMessageConsole.java]: Actually I think a better fix here is to construct the File Object, and then check that file.getCanonicalPath().startsWith(tmpDir) - if not then either throw an exception or if we want to support ".." then fall back to the getLogFlag() that you implemented above

Comment (sbliu) [reply main/plugins/org.talend.mdm.workbench.serverexplorer/src/main/java/org/talend/mdm/workbench/serverexplorer/console/MDMServerMessageConsole.java]: I think judge filename contains dot is more straightforward, but i can change to that.

File: main/plugins/org.talend.mdm.workbench.serverexplorer/src/main/java/org/talend/mdm/workbench/serverexplorer/console/MDMServerMessageConsole.java [hunk 2]
```diff
@@ -516,6 +516,11 @@ private void download(String dirPath, boolean needOpen, IProgressMonitor monitor
                 is = response.getEntity().getContent();
                 monitor.worked(60);
                 File file = new File(dirPath + File.separator + fileName);
+                if (!file.getCanonicalPath().startsWith(dirPath)) {
+                    fileName = getLogFlag() + ".log";
```
Comment (coheigea) [inline main/plugins/org.talend.mdm.workbench.serverexplorer/src/main/java/org/talend/mdm/workbench/serverexplorer/console/MDMServerMessageConsole.java]: Looks good, but could you log a warning for this case?

Comment (sbliu) [reply main/plugins/org.talend.mdm.workbench.serverexplorer/src/main/java/org/talend/mdm/workbench/serverexplorer/console/MDMServerMessageConsole.java]: Added

**Pull request merged**